resolver/src/resolver_future.rs

```diff
@@ -588,6 +594,43 @@ mod tests {
         }
     }
 
+    #[test]
```

Comment:
I think you should add a test that would fail with the old behavior of `local.` but pass with the new behavior.

I think we should have more tests that verify that the search support works as Kubernetes expects; I am happy to contribute that test. I filed #415 to track that work.